segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -164,8 +169,18 @@ public PravegaRequestProcessor(StreamSegmentStore segmentStore, ServerConnection
     public void readSegment(ReadSegment readSegment) {
         Timer timer = new Timer();
         final String segment = readSegment.getSegment();
+
+        if (this.tokenVerifier != null && !tokenVerifier.verifyToken(segment,
```

Comment:
Can we factor this common code out into a method and call it every time. Looks like a lot of similar code.

Reply:
Ping!